src/shogun/clustering/KMeansBase.cpp

```diff
@@ -130,6 +130,44 @@ void KMeansBase::compute_cluster_variances()
 	}
 }
 
+SGMatrix<float64_t> KMeansBase::compute_std_dev() const
+{
+	require(cluster_centers.size() > 0, "KMeans is not trained!");
+
+	SGMatrix<float64_t> points = distance->get_rhs()
+	                                 ->as<DenseFeatures<float64_t>>()
+	                                 ->get_feature_matrix();
+	SGVector<float64_t> cluster_assignments = const_cast<KMeansBase*>(this)
+	                                              ->apply()
+	                                              ->as<MulticlassLabels>()
+	                                              ->get_labels();
+
+	SGVector<int32_t> counts(k);
+	SGMatrix<float64_t> means = cluster_centers.clone();
+	SGMatrix<float64_t> squares_sums(dimensions, k);
+
+	for (int32_t point_number : range(cluster_assignments.vlen))
+	{
+		auto cluster_number = (int32_t) cluster_assignments[point_number];
+		auto point = points.get_column(point_number);
+		auto& count = counts[cluster_number];
+		auto mean = means.get_column(cluster_number);
```

Comment:
const auto&

Reply:
The mean is not const https://github.com/shogun-toolbox/shogun/pull/5013/files/962cb88418495fd2ee9c45e08aae4da137ee9c19#diff-fe9a7913331223f6e47670e65015ea83R159

Reply:
Oh I see, sorry!